lib/controllers/journal.js

```diff
@@ -147,8 +151,19 @@ journal.get("/side-effects", guard(["admin", "programAdministrator", "clinician"
 
 const patientUserAndFreeTextNote = (req) => req.authClaims.scopes.every((scope) => scope === "user") && req.data && req.data.text;
 
+//Helper function to redirect to special helpDesk based rotues with special auth
+function redirectIfHelpDesk(localUrl) {
+    return function (req, res, next) {
+      if(req.user.role === "helpDesk") {
+        return res.redirect(307, localUrl);
+      } else {
+        return next();
+      }
+    };
+}
+
```

Comment:
I'm guessing this is because we need help desk to be able to post only journal entries that are notes, and not other types of journal entires. Yes?

Reply:
The more I think about it, however, it seems this is unnecessary.

Reply:
Yeah this allows us to have very different specific middleware for the route that helpdesk has access to. I approve 👍